### ISSUE TITLE
Rule Request: Newline between function, array, dictionary, tuple parameters

### ISSUE THREAD

**Issue by kimdv** [realm/SwiftLint#5564]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

After I read this [proposal](https://github.com/apple/swift-evolution/pull/2344) I tought it would be nice to have a rule like "if there is more than x parameters in a function/tuple/array/dictionary there must be a newline between".

This rule can also be implemented without this proposal implemented. 

The reason for this is makes it easier to review code. So the rule should be opt-in.
There could also be a configuration to state how many parameters there must be before newline is required.

It could also say that the trailing `)` should be on a newline. 
We could also have a configuration that require the `(` to have a trailing newline

Examples:
```swift
// Not OK
func foo(bar: Bar, baz: Baz)
func foo(
  bar: Bar,
  baz: Baz)

// OK
func foo(bar: Bar)
func foo(
  bar: Bar,
  baz: Baz
)
```

Not sure if it should be the same rule, but when calling a function we could also require newline.

Examples:
```swift
// Not OK
someClass.foo(bar: "bar", baz: "baz")
someClass.foo(
  bar: "bar",
  baz: "baz")

// OK
someClass.foo(bar: "bar")
someClass.foo(
  bar: "bar",
  baz: "baz"
)
```

**Comment by kimdv** [comment]
If it makes sense and it's a good idea I'm willing to try to implement it

**Comment by kimdv** [comment]
Is already in SwiftLint
